### PR TITLE
fix: pad/truncate GPT batch to seq_length when pipeline parallel is enabled

### PR DESCRIPTION
**Pull request by yaoyu-33**
Problem
- Pipeline parallel expects static input shapes, but gpt_step.get_batch() did not enforce seq_length.

Fix
- When cfg.model.pipeline_model_parallel_size > 1, pad/truncate tokens/labels/loss_mask/position_ids (and attention_mask if present) to cfg.model.seq_length.

Files
- src/megatron/bridge/training/gpt_step.py

### CODE REVIEW

File: src/megatron/bridge/training/gpt_step.py [hunk 1]
```diff
@@ -121,14 +126,24 @@ def get_batch(
     # slice batch along sequence dimension for context parallelism
     batch = get_batch_on_this_cp_rank(batch)
 
+    # When using pipeline parallelism, ensure fixed shapes equal to cfg.model.seq_length
+    # (PP expects static shapes for inputs on first stage and labels/loss on last stage).
+    if getattr(cfg.model, "pipeline_model_parallel_size", 1) > 1:
+        seq_len = cfg.model.seq_length
+        batch["tokens"] = pad_or_truncate_2d_to_len(batch.get("tokens"), seq_len, seq_len, pad_value=0)  # type: ignore[assignment]
+        batch["labels"] = pad_or_truncate_2d_to_len(batch.get("labels"), seq_len, seq_len, pad_value=-100)  # type: ignore[assignment]
```
Comment (cuichenx) [inline src/megatron/bridge/training/gpt_step.py:134]: where is the -100 value coming from?

File: src/megatron/bridge/training/gpt_step.py [hunk 2]
```diff
@@ -121,14 +126,24 @@ def get_batch(
     # slice batch along sequence dimension for context parallelism
     batch = get_batch_on_this_cp_rank(batch)
 
+    # When using pipeline parallelism, ensure fixed shapes equal to cfg.model.seq_length
+    # (PP expects static shapes for inputs on first stage and labels/loss on last stage).
+    if getattr(cfg.model, "pipeline_model_parallel_size", 1) > 1:
+        seq_len = cfg.model.seq_length
+        batch["tokens"] = pad_or_truncate_2d_to_len(batch.get("tokens"), seq_len, seq_len, pad_value=0)  # type: ignore[assignment]
```
Comment (cuichenx) [inline src/megatron/bridge/training/gpt_step.py:133]: is this safe for packed sequence training?

**Comment by yaoyu-33** [comment]
need to verify seqpacking and thd before merge
